src/ekka_autocluster.erl

```diff
@@ -160,6 +164,39 @@ find_oldest_node(Nodes) ->
             ?LOG(error, "Bad nodes found: ~p, ResL: ", [BadNodes, ResL]), false
    end.
 
+%% @doc Core node discovery used by mria by replicant nodes to find
+%% the core ones.
+-spec(core_node_discovery_callback() -> [node()]).
+core_node_discovery_callback() ->
+    with_strategy(
+      fun(Mod, Opts) ->
+              case Mod:discover(Opts) of
+                  {ok, Nodes} ->
+                      filter_core_nodes(Nodes);
+                  {error, Reason} ->
+                      ?LOG(error, "Core node discovery error: ~p", [Reason]),
+                      []
+              end
+      end).
+
+filter_core_nodes(Nodes) ->
```

Comment:
I think it duplicates the logic of `mria_lb:check_same_cluster` that filters the node list. Perhaps the additioanal checks can be skipped in ekka?